packages/core/src/boot/init.ts

```diff
@@ -48,6 +49,7 @@ export interface BuildEnv {
 }
 
 export function commonInit(initConfiguration: InitConfiguration, buildEnv: BuildEnv) {
+  updateExperimentalFeatures(initConfiguration.enableExperimentalFeatures || [])
```

Comment:
do we need to default to `[]` here?

Reply:
FMU,
Since the FF data is arriving externally from a JS snippet, we should have minimum validation to ensure we have an array since TS won't help runtime issues. To handle the case of being possibly undefined, we can either have this inside the TS file, or here. To me it makes sense to trust TS checking inside the func, so we should default to array outside the func.

Reply:
I though you were handling the case where no flags were provided here.

> To me it makes sense to trust TS checking inside the func

you are already doing some sanitization inside the function `Array.isArray` and `.filter((flag) => typeof flag === 'string')` which make me think you don't really trust the `enabledFeatures: string[]` typing.

what about doing all the sanitization at the same place?

Reply:
1. enableExperimentalFeatures is typed within initConfig as 
`enableExperimentalFeatures?: string[] | undefined`
so we handle the undefined case outside the `updateExperimentalFeatures` with `initConfiguration.enableExperimentalFeatures || []`
2. Inside updateExperimentalFeatures, as an API, we IMO, shouldn't allow the undefined case in the context of typescript checking because we should be passing string[] only.
3. We can't depend on the runtime following the TS rules given customer control, so we handle them now inside `updateExperimentalFeatures` (changed from inside `commonInit` as a previous commit in this PR ).

Do you have an alternative in mind?

Reply:
> 1. enableExperimentalFeatures is typed within initConfig as
enableExperimentalFeatures?: string[] | undefined
so we handle the undefined case outside the updateExperimentalFeatures with initConfiguration.enableExperimentalFeatures || []
> 2. Inside updateExperimentalFeatures, as an API, we IMO, shouldn't allow the undefined case in the context of typescript checking because we should be passing string[] only.

why though?
do we have other use cases for `updateExperimentalFeatures` where we should not allow to pass `undefined`?

IMO, we could go with `updateExperimentalFeatures(initConfiguration.enableExperimentalFeatures)` or even `updateExperimentalFeatures(initConfiguration)` since we only use it from the `initConfiguration`.

it would allow us to handle the undefined case in the unit tests as well.

Reply:
Would it be a valid solution to have a `sanitise` for the entire `initConfiguration`? This way we ensure to have a valid typescript object before any other logic.

Reply:
Cool, switched to `updateExperimentalFeatures(initConfiguration.enableExperimentalFeatures)`